agent/app/agent_capability_unix.go

```diff
@@ -88,6 +93,52 @@ func (agent *ecsAgent) appendENITrunkingCapabilities(capabilities []*ecs.Attribu
 	return agent.appendBranchENIPluginVersionAttribute(capabilities)
 }
 
+func lineCounter(r io.Reader) (int, error) {
```

Comment:
I think [Scanner](https://golang.org/pkg/bufio/#Scanner) will do what you are doing here.